fenix/app/src/main/java/org/mozilla/fenix/extension/WebExtensionPromptFeature.kt

```diff
@@ -79,13 +106,76 @@ class WebExtensionPromptFeature(
 
     private fun handlePermissionRequest(
         addon: Addon,
-        promptRequest: WebExtensionPromptRequest.Permissions,
+        promptRequest: WebExtensionPromptRequest.AfterInstallation.Permissions,
     ) {
-        if (hasExistingPermissionDialogFragment()) {
-            return
+        if (hasExistingPermissionDialogFragment()) return
+        showPermissionDialog(
+            addon,
+            promptRequest,
+        )
+    }
+
+    @VisibleForTesting
+    internal fun handleInstallationFailedRequest(
+        exception: WebExtensionInstallException,
+    ) {
+        val addonName = exception.extensionName ?: ""
+        var title = context.getString(R.string.mozac_feature_addons_failed_to_install, "")
+        val message = when (exception) {
+            is WebExtensionInstallException.Blocklisted -> {
+                context.getString(R.string.mozac_feature_addons_blocklisted, addonName)
+            }
+
+            is WebExtensionInstallException.Unknown, is WebExtensionInstallException.UserCancelled -> {
+                // Making sure we don't have a
+                // Title = Failed to install
+                // Message = Failed to install $addonName
+                title = ""
+                context.getString(R.string.mozac_feature_addons_failed_to_install, addonName)
+            }
+
+            is WebExtensionInstallException.NetworkFailure -> {
+                context.getString(R.string.mozac_feature_addons_failed_to_install_network_error)
+            }
+
+            is WebExtensionInstallException.CorruptedFile -> {
+                context.getString(R.string.mozac_feature_addons_failed_to_install_corrupt_error)
+            }
+
+            is WebExtensionInstallException.IncorrectHash -> {
+                context.getString(R.string.mozac_feature_addons_failed_to_install_hash_error)
+            }
```

Comment:
I don't think we can easily run into this error since this seems to be only used for updates, which aren't triggered by users. Maybe we could omit this error. WDYT?

Reply:
Removed :)